nemo/collections/common/parts/preprocessing/manifest.py

```diff
@@ -61,17 +67,41 @@ def item_iter(
     if isinstance(manifests_files, str):
         manifests_files = [manifests_files]
 
+    num_manifests = len(manifests_files)
+    if isinstance(data_prefix, str):
+        # use the same prefix for all manifests
+        data_prefix = [data_prefix] if num_manifests == 1 else [data_prefix]*num_manifests
+    elif data_prefix is None:
+        # append no prefix
+        data_prefix = [""] * num_manifests
+    
+    num_prefix = len(data_prefix)
+    if num_prefix > 1 and num_prefix != num_manifests:
+        raise ValueError(
+            f"The number of manifests and prefixes must match, unless a single or none prefix is provided." \
+             f"Got {num_manifests} manifests with {num_prefix} prefix(es)."
+        )
+
     if parse_func is None:
         parse_func = __parse_item
 
     k = -1
-    for manifest_file in manifests_files:
+    for manifest_file, prefix in zip(manifests_files, data_prefix):
+        if not Path(manifest_file).is_file():
+            logging.info(f"Manifest not found: {manifest_file}")
+            manifest_file = str(Path(prefix) / Path(manifest_file))
+            logging.info(f"Try adding prefix {prefix}: {manifest_file}")
+            if Path(manifest_file).is_file():
+                logging.info(f"Found alternate manifest: {manifest_file}")
+            else:
+                raise ValueError(f"Manifest not found: {manifest_file}")
+
         with open(expanduser(manifest_file), 'r') as f:
             for line in f:
                 k += 1
                 item = parse_func(line, manifest_file)
                 item['id'] = k
-
+                item['audio_file'] = str(Path(prefix) / Path(item['audio_file']))
```

Comment:
Don't override value of output of parse_func, it will cause errors if output of func is unexpectedly post processed. See if you can modify the functions itself.